core/src/main/java/com/linecorp/armeria/server/RedirectService.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+
+/**
+ * An {@link HttpService} that implements an HTTP redirection.
+ *
+ * <p>This class will redirect requests from its bound {@link Service} path pattern
+ * to a new location pattern. Currently only these patterns are supported:
+ * <ul>
+ *   <li>{@code /new} (no path parameters)</li>
+ *   <li>{@code /new/{var1}} (curly-brace style parameters)</li>
+ *   <li>{@code /new/:var1/new/:var2} (colon style parameters)</li>
+ *   <li>{@code http://host/new} (full URL without path parameters)</li>
+ *   <li>{@code http://host/new/{var1}} (full URL with curly-brace style parameters)</li>
+ *   <li>{@code http://host/new/:var1/new/:var2} (full URL with colon style parameters)</li>
+ * </ul>
+ * The {@link RedirectService} will return {@link HttpStatus#TEMPORARY_REDIRECT 307 Temporary Redirect}
+ * by default.
+ * <pre>{@code
+ * ServerBuilder sb = ...;
+ * sb.service("/old/", new RedirectService("/new"));
```

Comment:
`("/new")));` ?? and we need to add one more `)` for all of the following examples.

Reply:
It seems like two closing parentheses are correct here. I see only one closing parenthesis for some of the example below though.

Reply:
Oops my bad~